src/background/services/secrets/SecretsService.ts

```diff
@@ -146,23 +167,30 @@ export class SecretsService {
     }
   }
 
-  getActiveWalletSecrets(walletKeys: WalletSecretInStorage) {
-    const accountsService = container.resolve(AccountsService);
-
-    const activeWalletId = isPrimaryAccount(accountsService.activeAccount)
-      ? accountsService.activeAccount.walletId
-      : accountsService.activeAccount?.id;
+  getActiveWalletSecrets(
+    walletKeys: WalletSecretInStorage,
+    activeAccount: Account
+  ) {
```

Comment:
Similar comment to the one about `getActiveAccountSecrets` method - this method is more generic now, we can pass any account and therefore retrieve any wallet's secrets:

```suggestion
  getWalletSecretsForAccount(
    walletKeys: WalletSecretInStorage,
    account: Account
  ) {
```